src/utils/flags.ts

```diff
@@ -33,11 +40,115 @@ const parsePathInfo = async (input: string, exists: boolean): Promise<PathInfo>
-): Interfaces.OptionFlag<PathInfo | undefined> => {
+type EnsureFsFlagOptions = {
+  flagName: string;
+  path: string;
+  type: 'dir' | 'file' | 'any';
+  throwOnENOENT?: boolean;
+};
+
+interface FsError extends Error {
+  code: string;
+}
+
+/**
+ * Ensures command flags that are file system paths are set properly before
+ * continuing command execution.  Can also create directories that don't yet
+ * exist in the path.
+ *
+ * @param options defines the path to resolve and the expectations
+ * @returns the resolved flag path
+ */
+function ensureFlagPath(options: EnsureFsFlagOptions): string {
+  const { flagName, path, type, throwOnENOENT } = options;
+
+  const trimmedPath = path?.trim();
+  let resolvedPath: string;
+  if (trimmedPath?.length) {
+    resolvedPath = resolve(trimmedPath);
+  }
+
+  try {
+    const stats = fs.statSync(resolvedPath);
+    if (type !== 'any') {
+      const isDir = stats.isDirectory();
+      if (type === 'dir' && !isDir) {
+        const msg = messages.getMessage('error.ExpectedDirectory');
+        throw new SfError(messages.getMessage('error.InvalidFlagPath', [flagName, path, msg]), 'InvalidFlagPath');
+      } else if (type === 'file' && isDir) {
+        const msg = messages.getMessage('error.ExpectedFile');
+        throw new SfError(messages.getMessage('error.InvalidFlagPath', [flagName, path, msg]), 'InvalidFlagPath');
+      }
+    }
+  } catch (error: unknown) {
+    const err = error as FsError;
+    if (err.code !== 'ENOENT') {
+      throw err;
+    } else {
+      if (throwOnENOENT) {
+        const enoent = messages.getMessage('error.PathNotFound');
+        throw new SfError(messages.getMessage('error.InvalidFlagPath', [flagName, path, enoent]), 'InvalidFlagPath');
+      }
+      const dir = type === 'dir' ? resolvedPath : dirname(resolvedPath);
+      fs.mkdirSync(dir, { recursive: true });
+    }
+  }
+  return resolvedPath;
+}
+
+/**
+ * Flag value is a directory path that may or may not exist. If it doesn't exist, then it will be created.
+ */
+export const ensuredDirFlag = (opts: EnsuredDirOpts): Interfaces.OptionFlag<string | undefined> => {
+  return Flags.build<string | undefined>({
+    parse: async (input: string) => {
+      return Promise.resolve(
+        ensureFlagPath({
+          flagName: opts.name,
+          path: input,
+          type: 'dir',
+        })
+      );
+    },
+    ...opts,
+  })();
+};
+
+function resolveZipFileName(zipFileName?: string): string {
+  // If no file extension was provided append, '.zip'
+  if (zipFileName && !extname(zipFileName)) {
+    zipFileName += '.zip';
+  }
+  return zipFileName || 'unpackaged.zip';
+}
+
+export const testLevelFlag = (
+  opts: Partial<Interfaces.OptionFlag<TestLevel | undefined>> = {}
+): Interfaces.OptionFlag<TestLevel | undefined> => {
+  return Flags.build<TestLevel | undefined>({
+    char: 'l',
+    parse: (input: string) => Promise.resolve(input as TestLevel),
```

Comment:
What happens without the parse function that's doing nothing (it's either undefined or one of the options!?). Or is it necessary to coerce the type via the assertion?

Reply:
`Flags.build` requires the `parse` to be present so you have to put something.

If you did `parse: (input: string) => Promise.resolve(input)`, it would complain that you're returning a string when it expects a `TestLevel`

Reply:
that feels like an oclif problem.